libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -57,11 +68,37 @@
  */
 #define MQTT_DISCONNECT_REMAINING_LENGTH    ( ( uint8_t ) 0 )
 
+/*
+ * Constants relating to CONNACK packets, defined by MQTT 3.1.1 spec.
+ */
+#define MQTT_PACKET_CONNACK_REMAINING_LENGTH        ( ( uint8_t ) 2U )    /**< @brief A CONNACK packet always has a "Remaining length" of 2. */
```

Comment:
Why are we doing a #define here? The magic numbers smell applies to numbers that we may want to change. This is mandated by the spec so it can never change. It would be better to just have a comment where we check the length.

Reply:
After (offline?) discussion, will leave the macros here and add comments in code for readability